com.unity.netcode.gameobjects/Runtime/Serialization/FastBufferReader.cs

```diff
@@ -0,0 +1,987 @@
+                        value = Enum.ToObject(type, ulongVal);
+                        return;
+                }
+            }
+
+            if (type == typeof(GameObject))
+            {
+                ReadValueSafe(out GameObject go);
+                value = go;
+                return;
+            }
+
+            if (type == typeof(NetworkObject))
+            {
+                ReadValueSafe(out NetworkObject no);
+                value = no;
+                return;
+            }
+
+            if (typeof(NetworkBehaviour).IsAssignableFrom(type))
+            {
+                ReadValueSafe(out NetworkBehaviour nb);
+                value = nb;
+                return;
+            }
+            /*if (value is INetworkSerializable)
+            {
+                //TODO ((INetworkSerializable)value).NetworkSerialize(new NetworkSerializer(this));
+                return;
+            }*/
+
+            throw new ArgumentException($"{nameof(FastBufferReader)} cannot read type {type.Name} - it does not implement {nameof(INetworkSerializable)}");
+        }
+
+        /// <summary>
+        /// Read an INetworkSerializable
+        /// </summary>
+        /// <param name="value">INetworkSerializable instance</param>
+        /// <typeparam name="T"></typeparam>
+        /// <exception cref="NotImplementedException"></exception>
+        public void ReadNetworkSerializable<T>(out T value) where T : INetworkSerializable
+        {
+            throw new NotImplementedException();
+        }
+
+        /// <summary>
+        /// Read a GameObject
+        /// </summary>
+        /// <param name="value">value to read</param>
+        public void ReadValue(out GameObject value)
+        {
+            ReadValue(out ulong networkObjectId);
+
+            if (NetworkManager.Singleton.SpawnManager.SpawnedObjects.TryGetValue(networkObjectId, out NetworkObject networkObject))
+            {
+                value = networkObject.gameObject;
+                return;
+            }
+
+            if (NetworkLog.CurrentLogLevel <= LogLevel.Normal)
+            {
+                NetworkLog.LogWarning($"{nameof(FastBufferReader)} cannot find the {nameof(GameObject)} sent in the {nameof(NetworkSpawnManager.SpawnedObjects)} list, it may have been destroyed. {nameof(networkObjectId)}: {networkObjectId}");
+            }
+
+            value = null;
+        }
+
+        /// <summary>
+        /// Read a GameObject
+        ///
+        /// "Safe" version - automatically performs bounds checking. Less efficient than bounds checking
+        /// for multiple reads at once by calling VerifyCanRead.
+        /// </summary>
+        /// <param name="value">value to read</param>
+        public void ReadValueSafe(out GameObject value)
+        {
+            ReadValueSafe(out ulong networkObjectId);
+
+            if (NetworkManager.Singleton.SpawnManager.SpawnedObjects.TryGetValue(networkObjectId, out NetworkObject networkObject))
```

Comment:
This singleton use here will cause issues with multi instance test where this will always reference the same instance of a NetworkObject instead of the correct one.

Reply:
Same goes for NetworkObject, NetworkBehaviour read/writes. Could we consider scrapping them until we have a need for them and then implement a better solution?